app/src/main/java/com/zulip/android/activities/DevAuthActivity.java

```diff
@@ -0,0 +1,147 @@
+
+import android.app.Activity;
+import android.app.ProgressDialog;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.zulip.android.R;
+import com.zulip.android.networking.AsyncLogin;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DevAuthActivity extends Activity {
+    RecyclerView recyclerView;
+    private ProgressDialog connectionProgressDialog;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_dev_auth);
+        recyclerView = (RecyclerView) findViewById(R.id.devAuthRecyclerView);
+        String json = getIntent().getStringExtra("emails_json");
+        List<String> emails = new ArrayList<>();
+        int direct_admin_size = 1;
+        connectionProgressDialog = new ProgressDialog(this);
+        connectionProgressDialog.setMessage(getString(R.string.signing_in));
+
+        try {
+            JSONObject jsonObject = new JSONObject(json);
+            JSONArray jsonArray = jsonObject.getJSONArray("direct_admins");
+            for (int i = 0; i < jsonArray.length(); i++) {
+                emails.add(jsonArray.get(i).toString());
+            }
+            direct_admin_size = jsonArray.length();
+            jsonArray = jsonObject.getJSONArray("direct_users");
+            for (int i = 0; i < jsonArray.length(); i++) {
+                emails.add(jsonArray.get(i).toString());
+            }
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+        AuthEmailAdapter authEmailAdapter = new AuthEmailAdapter(emails, direct_admin_size);
+        recyclerView.setAdapter(authEmailAdapter);
+        recyclerView.setLayoutManager(new LinearLayoutManager(DevAuthActivity.this) {
+        });
+
+
+    }
+
+    public void loginThroughEmail(String email) {
+        AsyncLogin asyncLogin = new AsyncLogin(DevAuthActivity.this, email, null, true);
+        asyncLogin.execute();
+        connectionProgressDialog.show();
+    }
+
+    public void openHome() {
+        // Cancel before leaving activity to avoid leaking windows
+        connectionProgressDialog.dismiss();
+        Intent i = new Intent(this, ZulipActivity.class);
+        startActivity(i);
+        finish();
+    }
+    public class AuthEmailAdapter extends RecyclerView.Adapter<DevAuthActivity.AuthEmailViewHolder> {
+
+        private static final int VIEW_TYPE_HEADER = 0;
+        private static final int VIEW_TYPE_ADMIN = 1;
+        private static final int VIEW_TYPE_USER = 2;
+        List<String> emails;
+        int direct_admin_size;
```

Comment:
this style of variable naming is not great. please use camel casing. ie `directAdminSize`.